src/effects/effectbuttonparameterslot.h

```diff
@@ -29,23 +29,14 @@ class EffectButtonParameterSlot : public EffectParameterSlotBase {
 
     void setValue(double value);
 
-    // Clear the currently loaded effect
-    void clear();
-
     QDomElement toXml(QDomDocument* doc) const override;
     void loadParameterSlotFromXml(const QDomElement& parameterElement) override;
 
-  public slots:
-    void updateEngineState();
-
   private:
     QString debugString() const {
         return QString("EffectButtonParameterSlot(%1,%2)").arg(m_group).arg(m_iParameterSlotNumber);
     }
 
-    // Control exposed to the rest of Mixxx
-    ControlPushButton* m_pControlValue;
```

Comment:
Correct me if I'm wrong, but I think you still need this to access the methods of ControlPushButton in EffectButtonParameterSlot that are not common to ControlObject.

Reply:
This is required to move updateEngineState to EffectParameterSlotBase. We can use ControlPushButton specific methods by typecasting.